projects/rocprofiler-systems/source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

```diff
@@ -577,13 +577,32 @@ cache_kernel_dispatch(rocprofiler_buffer_tracing_kernel_dispatch_record_t* recor
         record->dispatch_info.grid_size.z, stream_handle });
 }
 
+void
+cache_scratch_memory(rocprofiler_buffer_tracing_scratch_memory_record_t* record,
+                     uint64_t                                            stream_handle)
+{
+    trace_cache::get_metadata_registry().add_stream(stream_handle);
+    trace_cache::get_buffer_storage().store(trace_cache::scratch_memory_sample{
+        record->start_timestamp, record->end_timestamp, record->thread_id,
+        record->agent_id.handle, record->queue_id.handle,
+        static_cast<int32_t>(record->kind), static_cast<int32_t>(record->operation),
+        static_cast<int32_t>(record->flags),
+// Scratch memory samples from SDK versions prior to 7.0.2 do not include allocation_size
+#if(ROCPROFSYS_USE_ROCM > 0 && ROCPROFSYS_ROCM_VERSION >= 70002)
+        record->allocation_size,
+#else
+        0,
+#endif
```

Comment:
Maybe you can extract this to separate function, something like:

```cpp
auto get_scratch_memory_allocation_size(rocprofiler_buffer_tracing_scratch_memory_record_t* record)
{
// Scratch memory samples from SDK versions prior to 7.0.2 do not include allocation_size
#if(ROCPROFSYS_USE_ROCM > 0 && ROCPROFSYS_ROCM_VERSION >= 70002)
     return   record->allocation_size;
#else
        return 0;
#endif
}
```

and just to call it here.
```suggestion
get_scratch_memory_allocation_size(record);
```


We will get cleaner code, and this ifdef logic will be isolated.

Reply:
Done, thank you!